api/utils/rebate_checker.py

```diff
@@ -0,0 +1,20 @@
+from django.utils import timezone
+from home.models.students import Student, LongRebate, TodayRebate, Allocation
+
+
+def is_student_on_rebate(student: Student, allocation: Allocation):
+    """
+    This function checks if the student is on rebate or not
+    """
+    long_rebate = LongRebate.objects.filter(email=student)
```

Comment:
we can use less than and greater than checks while querying itself. this will reduce the in-memory items we will be having.
https://docs.djangoproject.com/en/5.1/ref/models/querysets/